### DISCUSSION TITLE
Support .css in node_modules (possibly using "next/dynamic")

### DISCUSSION THREAD

**Discussion by hcharley** [vercel/next.js#25483] [Ideas]
### Describe the feature you'd like to request

I would like to be able import CSS from node_modules just as I might from my own code. 

In this case, it's my own code in node_modules. 

A core module publishs my `<RichTextEditor />` component which is a thin `next/dynamic` wrapper:

```tsx
import dynamic from 'next/dynamic';

export const RichTextEditor = dynamic(
  () => import('./rich-text-editor-component'),
  {
    ssr: false,
  }
);
```

```tsx
import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css';

export const RichTextEditorInner = ({ defaultValue, onChange }: Props) => {
...
}
```

Instead of exporting it directly and incurring increased package sizes, I just was hoping to export the dynamic component.

### Describe the solution you'd like

I'd ideally like to specify how to handle node_module imports of css and overcome the problems listed in the next documentation:

> Possible Ways to Fix It
> Reach out to the maintainer and ask for them to publish a compiled version of their dependency.
> 
> Compiled dependencies do not have references to CSS files, or any other files that require bundler-specific integrations.
> 
> The dependency should also provide instructions about what CSS needs to be imported by you, in your application.
> 
> Importing CSS files within node_modules cannot be supported because we cannot know the correct behavior:
> 
> Should the file be consumed as Global CSS or CSS Modules?
> If Global, in what order does the file need to be injected?
> If Modules, what are the emitted class names? As-is, camel-case, snake case?
> Etc...
> 

https://nextjs.org/docs/messages/css-npm

### Describe alternatives you've considered

One way I've considered is only doing so at the `next/dynamic` level:

```tsx
import dynamic from 'next/dynamic';

export const RichTextEditor = dynamic(
  () => import('./rich-text-editor-component'),
  {
    ssr: false,
    importCssMode: 'module' | 'global',
    globalCssInjectionOrder: '100%' | 1 | 0.5 | '50%' | 0.25 | '25%' ... // 100% could mean at the end, 25% could be inserted in spot after the 2.5th item of a list of 10, or 5
0% position as 5th of a list of 10, ...
    cssModuleCasingStrategy: 'camelCase' | 'snakeCase'
  }
);
```

I think you could probably get away with hardcoding `globalCssInjectionOrder` to the last spot to start with, since it seems to make sense that components being imports last dynamically would want their styles to override anything above it. Next could keep track of dynamic import orders and then just keep the global CSS in that order.

**Comment by timneutkens** [comment]
Opened a RFC here: #27953
